core/server.go

```diff
@@ -299,8 +299,16 @@ func (m *RpcServer) DestroyEnvironment(cxt context.Context, req *pb.DestroyEnvir
 		return nil, status.Newf(codes.NotFound, "environment not found: %s", err.Error()).Err()
 	}
 
-	statesForDestroy := [...]string{"CONFIGURED", "STANDBY"}
+	if req.AllowInRunningState && env.CurrentState() == "RUNNING" {
+		err = env.TryTransition(environment.MakeTransition(m.state.taskman, pb.ControlEnvironmentRequest_CONFIGURE))
```

Comment:
This should be a `ControlEnvironmentRequest_STOP_ACTIVITY`, otherwise the transition will necessarily fail.

Reply:
👍